addons/calendar/models/calendar_alarm_manager.py

```diff
@@ -153,7 +152,7 @@ def _get_events_by_alarm_to_notify(self, alarm_type):
         already.
         """
         lastcall = self.env.context.get('lastcall', False) or fields.date.today() - relativedelta(weeks=1)
-        now = datetime.now(tz=UTC)
+        now = fields.Datetime.now()
```

Comment:
>[!note]
> This change was needed, as formatting `datetime.now(tz=UTC)` into the SQL query turns it into a `timestamp with time zone`, meaning it will rotate the `timestamp without time zone` values of `calendar_event` using the database's `TimeZone` configuration parameter[^1] when comparing.
> Instead we can just use `fields.Datetime.now()` which ensures the value is suitable for the ORM.

[^1]: https://www.postgresql.org/docs/15/functions-datetime.html